src/libraries/System.Threading.RateLimiting/src/System/Threading/RateLimiting/TokenBucketRateLimiterOptions.cs

```diff
@@ -0,0 +1,93 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Threading.RateLimiting
+{
+    /// <summary>
+    /// Options to control the behavior of a <see cref="TokenBucketRateLimiter"/>.
+    /// </summary>
+    public sealed class TokenBucketRateLimiterOptions
+    {
+        /// <summary>
+        /// Initializes the <see cref="TokenBucketRateLimiterOptions"/>.
+        /// </summary>
+        /// <param name="tokenLimit">Maximum number of tokens that can be in the token bucket.</param>
+        /// <param name="queueProcessingOrder"></param>
+        /// <param name="queueLimit">Maximum number of unprocessed tokens waiting via <see cref="RateLimiter.WaitAsync(int, CancellationToken)"/>.</param>
+        /// <param name="replenishmentPeriod">
+        /// Specifies how often tokens can be replenished. Replenishing is triggered either by an internal timer if <paramref name="autoReplenishment"/> is true, or by calling <see cref="TokenBucketRateLimiter.TryReplenish"/>.
+        /// </param>
+        /// <param name="tokensPerPeriod">Specified how many tokens can be added to the token bucket on a successful replenish. Available token count will not exceed <paramref name="tokenLimit"/>.</param>
+        /// <param name="autoReplenishment">
+        /// Specifies whether token replenishment will be handled by the <see cref="TokenBucketRateLimiter"/> or by another party via <see cref="TokenBucketRateLimiter.TryReplenish"/>.
+        /// </param>
+        /// <exception cref="ArgumentOutOfRangeException">When <paramref name="tokenLimit"/>, <paramref name="queueLimit"/>, or <paramref name="tokensPerPeriod"/> are less than 0
+        /// or when <paramref name="replenishmentPeriod"/> is more than 49 days.</exception>
+        public TokenBucketRateLimiterOptions(
+            int tokenLimit,
+            QueueProcessingOrder queueProcessingOrder,
+            int queueLimit,
+            TimeSpan replenishmentPeriod,
+            int tokensPerPeriod,
+            bool autoReplenishment = true)
+        {
+            if (tokenLimit < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(tokenLimit));
+            }
+            if (queueLimit < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(queueLimit));
+            }
+            if (tokensPerPeriod < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(tokensPerPeriod));
+            }
+            if (replenishmentPeriod.TotalDays > 49)
+            {
+                throw new ArgumentOutOfRangeException(nameof(replenishmentPeriod), "Over 49 days is not supported");
```

Comment:
Is this to account for the tick count wrapping around? A comment might be useful to explain why 49 days is a magic number.